src/main/scala/ai/datahunters/md/processor/FlattenMetadataTags.scala

```diff
@@ -32,11 +33,19 @@ case class FlattenMetadataTags(colPrefix: String,
 
     val flattenDF = inputDF.withColumn(MetadataCol, selectMetadataTagsUDF(col(MetadataCol)))
       .select(columns.head, columns.tail:_*)
-    val transformedDF = if (includeMetadataContent) {
-      flattenDF.withColumn(MetadataContentCol, concat_ws(" ", buildConcatTagList(selectedTags, colPrefix):_*))
+    val arrayFieldsToFlatten = selectArrayColumns(flattenDF)
+    val afterArraysFlatteningDF = if (flattenArrays) {
+      arrayFieldsToFlatten
+        .foldLeft(flattenDF)((previousDF, colName) => previousDF.withColumn(colName, concat_ws(",", col(colName))))
```

Comment:
Please consider to use FlattenArrays class, because implementation is very similar.

Reply:
@dawidsau I know but It's different processor, I thought about it but embedding processors could be confusing. Maybe better option would be to move somewhere logic out of processors and just call it inside both of them..